src/cgroups/v1/blkio.rs

```diff
@@ -33,46 +92,142 @@ impl Controller for Blkio {
     }
 }
 
+impl StatsProvider for Blkio {
+    type Stats = BlkioStats;
+
+    fn stats(cgroup_path: &Path) -> Result<Self::Stats> {
+        if cgroup_path.join(BLKIO_WEIGHT).exists() {
+            return Self::get_weight_division_policy_stats(cgroup_path);
+        }
+
+        Self::get_throttling_policy_stats(cgroup_path)
+    }
+}
+
 impl Blkio {
     fn apply(root_path: &Path, blkio: &LinuxBlockIo) -> Result<()> {
         for trbd in &blkio.blkio_throttle_read_bps_device {
             common::write_cgroup_file_str(
-                &root_path.join(CGROUP_BLKIO_THROTTLE_READ_BPS),
+                &root_path.join(BLKIO_THROTTLE_READ_BPS),
                 &format!("{}:{} {}", trbd.major, trbd.minor, trbd.rate),
             )?;
         }
 
         for twbd in &blkio.blkio_throttle_write_bps_device {
             common::write_cgroup_file_str(
-                &root_path.join(CGROUP_BLKIO_THROTTLE_WRITE_BPS),
+                &root_path.join(BLKIO_THROTTLE_WRITE_BPS),
                 &format!("{}:{} {}", twbd.major, twbd.minor, twbd.rate),
             )?;
         }
 
         for trid in &blkio.blkio_throttle_read_iops_device {
             common::write_cgroup_file_str(
-                &root_path.join(CGROUP_BLKIO_THROTTLE_READ_IOPS),
+                &root_path.join(BLKIO_THROTTLE_READ_IOPS),
                 &format!("{}:{} {}", trid.major, trid.minor, trid.rate),
             )?;
         }
 
         for twid in &blkio.blkio_throttle_write_iops_device {
             common::write_cgroup_file_str(
-                &root_path.join(CGROUP_BLKIO_THROTTLE_WRITE_IOPS),
+                &root_path.join(BLKIO_THROTTLE_WRITE_IOPS),
                 &format!("{}:{} {}", twid.major, twid.minor, twid.rate),
             )?;
         }
 
         Ok(())
     }
+
+    fn get_throttling_policy_stats(cgroup_path: &Path) -> Result<BlkioStats> {
+        let mut stats = BlkioStats::default();
+
+        stats.service_bytes =
+            Self::parse_blkio_file(&cgroup_path.join(BLKIO_THROTTLE_IO_SERVICE_BYTES))?;
+        stats.serviced = Self::parse_blkio_file(&cgroup_path.join(BLKIO_THROTTLE_IO_SERVICED))?;
```

Comment:
https://rust-lang.github.io/rust-clippy/master/index.html#field_reassign_with_default
```suggestion
        let stats = BlkioStats {
            service_bytes: Self::parse_blkio_file(
                &cgroup_path.join(BLKIO_THROTTLE_IO_SERVICE_BYTES),
            )?,
            serviced: Self::parse_blkio_file(&cgroup_path.join(BLKIO_THROTTLE_IO_SERVICED))?,
            ..Default::default()
        };
```